### PR TITLE
Fixes broken links in Documentation

### PR DESCRIPTION
**Pull request by jeevantelukula**
Fix broken links in the documentation for Migration Guides, How to Guides, Examples and Demos, Foundational Components Sections.

### CODE REVIEW

File: source/common/PRU-ICSS/PRU-Hands-on-Labs.rst [hunk 1]
```diff
@@ -60,7 +60,7 @@ Software
    However, the notation ***<PRU_SW_PATH>*** will be used throughout
    the labs to reference this assumed installation location. If you use
    a different location please modify the steps below.
--  `CCSv6 <http://processors.wiki.ti.com/index.php/Download_CCS#Code_Composer_Studio_Version_6_Downloads>`__
+-  `CCSv6 <ti.com/tool/download/CCSTUDIO>`__
```
Comment (StaticRocket) [inline source/common/PRU-ICSS/PRU-Hands-on-Labs.rst:63]: Keeping in line with the rest of these changes this URL should probably be a full `https://www.ti.com/...` target

File: source/common/PRU-ICSS/PRU-Hands-on-Labs.rst [hunk 2]
```diff
@@ -971,15 +970,8 @@ to your set up.
...
-
-          You can find instructions on this process at the
-          `Linux Create SD Card documentation <../../devices/AM64X/Overview/Create_SD_Card.html>`__.
-
-      .. ifconfig:: CONFIG_part_family not in ('AM64X_family')
-
-          You can find instructions on this process at the
-          `Linux SD Card Creation Guide <../../linux/Overview/Processor_SDK_Linux_create_SD_card_script.html>`__.
+      You can find instructions on this process at the
+      `Linux SD Card Creation Guide <../../linux/Overview/Processor_SDK_Linux_create_SD_card.html>`__.
```
Comment (StaticRocket) [inline source/common/PRU-ICSS/PRU-Hands-on-Labs.rst:974]: Please use a ref here. Interdocument HTML links are blocking #16

File: source/devices/AM335X/linux/Release_Specific_Migration_Guide.rst [hunk 3]
```diff
@@ -437,9 +432,7 @@ Processor SDK 03.03
 
 -  HDMI-connected monitor no longer takes precedence over built-in LCD
    as a default graphics display for e.g. Matrix
--  See the `Release
-   Notes <http://processors.wiki.ti.com/index.php/Processor_SDK_Linux_Release_Notes>`__
-   for full list of modifications.
+-  See the Release Notes for full list of modifications.
```
Comment (StaticRocket) [inline source/devices/AM335X/linux/Release_Specific_Migration_Guide.rst:435]: There's a ref for release notes you can use here

File: source/linux/Foundational_Components/U-Boot/UG-SATA.rst [hunk 4]
```diff
@@ -1,3 +1,5 @@
+.. _ug-sata:
```
Comment (StaticRocket) [inline source/linux/Foundational_Components/U-Boot/UG-SATA.rst:1]: If you just want to point at this doc you can use https://www.sphinx-doc.org/en/master/usage/referencing.html#cross-referencing-documents

**Review by StaticRocket — CHANGES_REQUESTED** [review]
A lot of these changes are useful. Thanks for taking the time to hilight these issues. Would you mind changing some of the inter-document links you added to use rst built in linking?

File: source/devices/AM335X/linux/Release_Specific_Migration_Guide.rst (hunk 3 above)
Comment (jeevantelukula) [reply source/devices/AM335X/linux/Release_Specific_Migration_Guide.rst:435]: But that release note corresponds to the release notes of older versions that are not available, so I removed the links.

File: source/common/EVM_Hardware_Setup/_EVMK2H_Hardware_Setup.rst [hunk 5]
```diff
@@ -493,14 +493,9 @@ Connect to EVMK2H using CCS
...
-below:
-
--  `Updating XDS2xx
-   Firmware <http://dev.ti.com/tirex/explore/node?node=AADzJ8Y-La4f7Bi5Ga0TcA__FUz-xrs__LATEST>`__
-
-.. note:: For newer versions of CCS, a XDS200 firmware update is recommended which user can perform using instruction on
-`Updating\_the\_XDS200\_firmware <http://dev.ti.com/tirex/explore/node?node=AADzJ8Y-La4f7Bi5Ga0TcA__FUz-xrs__LATEST>__`
+.. note::
+   For newer versions of CCS, a XDS200 firmware update is recommended which user can perform using instruction on the article
+   `Updating XDS200 Firmware <http://dev.ti.com/tirex/explore/node?node=AADzJ8Y-La4f7Bi5Ga0TcA__FUz-xrs__LATEST>`__
```
Comment (cshilwant) [inline source/common/EVM_Hardware_Setup/_EVMK2H_Hardware_Setup.rst:498]:
```suggestion
   `Updating XDS200 Firmware <https://dev.ti.com/tirex/explore/node?node=AADzJ8Y-La4f7Bi5Ga0TcA__FUz-xrs__LATEST>`__
```

File: source/devices/AM335X/linux/Release_Specific_Migration_Guide.rst [hunk 6]
```diff
@@ -552,10 +541,8 @@ Processor SDK 02.00.00
    integrated into Processor SDK, and supports null DRM based Full
    Window, with Front and Flip modes. For more information, see
    `Graphics
-   Migration <http://processors.wiki.ti.com/index.php/Processor_Linux_SDK_Graphics_and_Display#Migration_Guide_from_Processor_SDK_1.x_to_2.x_for_AM3.2C_AM4>`__
+   Migration <../../../linux/Foundational_Components/Graphics/SGX/Migration_From_Prior_Releases.html#from-processor-sdk-1-x-to-2-x-for-am3-am4>`__
```
Comment (cshilwant) [inline source/devices/AM335X/linux/Release_Specific_Migration_Guide.rst:544]: @jeevantelukula  Use ref tag here as well

File: source/linux/Examples_and_Demos/_Hands_on_with_QT.rst [hunk 7]
```diff
@@ -92,13 +93,12 @@ be adapted accordingly.
...
    -  For help creating a 2 partition SD card with the SDK content see
-      the `create\_sdcard.sh
-      script <Overview/Processor_SDK_Linux_create_SD_card_script.html>`__ page
+      the :ref:`create\_sdcard.sh script <processor-sdk-linux-sd-card-with-default-images>` page
 
 -  QT Creator 5.7.0 installed on your Linux host.
 
    -  You can download Qt Creator from open source distribution version
-      of Qt http://download.qt.io/official_releases/qt/5.7/5.7.0/
+      of Qt at http://download.qt.io/official_releases/qt/
```
Comment (cshilwant) [inline source/linux/Examples_and_Demos/_Hands_on_with_QT.rst:101]:
```suggestion
      of Qt at https://download.qt.io/official_releases/qt/
```

File: source/linux/Examples_and_Demos/_Hands_on_with_QT.rst [hunk 8]
```diff
@@ -1457,10 +1457,9 @@ style sheets.
 For debugging QT application with source code to QT libraries,
 corresponding QT library will need to be installed to the
 <ti-processor-sdk-linux-xxx>/linux-devkit/sysroots/armv7ahf-neon-linux-gnueabi
-location. The \*.ipk package can be found from the `yocto build of
-PSDK <http://processors.wiki.ti.com/index.php/Processor_SDK_Building_The_SDK>`__
-under
-build/arago-tmp-external-arm-toolchain/work/armv7ahf-neon-linux-gnueabi.
+location. The \*.ipk package can be found from the
+:ref:`yocto build of PSDK <building-the-sdk-with-yocto>`
+under build/arago-tmp-external-arm-toolchain/work/armv7ahf-neon-linux-gnueabi.
```
Comment (cshilwant) [inline source/linux/Examples_and_Demos/_Hands_on_with_QT.rst:1462]:
```suggestion
under :file:`build/arago-tmp-external-arm-toolchain/work/armv7ahf-neon-linux-gnueabi`.
```
https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#role-file

File: source/linux/Foundational_Components/_OpenCV.rst [hunk 9]
```diff
@@ -538,7 +538,7 @@ to AM57xx): **export OPENCV\_OPENCL\_DEVICE='TI AM57:ACCELERATOR:TI
 Multicore C66 DSP'**
 
 -  For additional information, please refer to:
-   `OpenCL User Guide <Foundational_Components_OpenCL.html>`__
+   `OpenCL User Guide <http://downloads.ti.com/mctools/esd/docs/opencl/index.html>`__
```
Comment (cshilwant) [inline source/linux/Foundational_Components/_OpenCV.rst:541]:
```suggestion
   `OpenCL User Guide <https://downloads.ti.com/mctools/esd/docs/opencl/index.html>`__
```

**Review by cshilwant — CHANGES_REQUESTED** [review]
Needs work!

File: source/devices/AM335X/linux/Release_Specific_Migration_Guide.rst (hunk 3 above)
Comment (StaticRocket) [reply source/devices/AM335X/linux/Release_Specific_Migration_Guide.rst:435]: If that content is truly unreachable then all references to it should be purged as well. Drop the note about " -  See the Release Notes for full list of modifications." as there's nothing to see.

**Review by cshilwant — APPROVED** [review]

**Review by StaticRocket — APPROVED** [review]

**Pull request merged**